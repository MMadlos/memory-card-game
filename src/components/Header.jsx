import pokeballLogo from "../assets/pokeball-logo.jpg"

export default function Header({ onClickInstructions }) {
	return (
		<div className="title-container">
			<img
				id="logo"
				src={pokeballLogo}
				alt="a normal pokeball as a logo of the page"
			/>
			<h1>Pokememory</h1>

			<div className="icon-container">
				<i
					className="fa-solid fa-circle-question"
					onClick={onClickInstructions}
				/>
				<a
					href="https://github.com/MMadlos/memory-card-game"
					target="_blank"
					rel="noreferrer">
					<i className="fa-brands fa-github" />
				</a>
			</div>
		</div>
	)
}
