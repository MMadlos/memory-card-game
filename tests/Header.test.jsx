import { render, screen } from "@testing-library/react"
import { describe, it, expect } from "vitest"

import Header from "../src/components/Header"

describe("Header component", () => {
	it("renders component", () => {
		const { container } = render(<Header />)
		expect(container).toMatchSnapshot()
	})

	it("renders correct heading", () => {
		render(<Header />)
		expect(screen.getByRole("heading").textContent).toMatch("Pokememory")
	})
})
