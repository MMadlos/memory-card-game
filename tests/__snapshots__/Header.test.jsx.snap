// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders component 1`] = `
<div>
  <div
    class="title-container"
  >
    <img
      alt="a normal pokeball as a logo of the page"
      id="logo"
      src="/src/assets/pokeball-logo.jpg"
    />
    <h1>
      Pokememory
    </h1>
    <div
      class="icon-container"
    >
      <i
        class="fa-solid fa-circle-question"
      />
      <a
        href="https://github.com/MMadlos/memory-card-game"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fa-brands fa-github"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Header component > renders title container 1`] = `
<div>
  <div
    class="title-container"
  >
    <img
      alt="a normal pokeball as a logo of the page"
      id="logo"
      src="/src/assets/pokeball-logo.jpg"
    />
    <h1>
      Pokememory
    </h1>
    <div
      class="icon-container"
    >
      <i
        class="fa-solid fa-circle-question"
      />
      <a
        href="https://github.com/MMadlos/memory-card-game"
        rel="noreferrer"
        target="_blank"
      >
        <i
          class="fa-brands fa-github"
        />
      </a>
    </div>
  </div>
</div>
`;
